### PR TITLE
misc engine changes

### PR DESCRIPTION
**Pull request by jamesemery**

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -44,7 +44,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
+DEFAULT_JVM_OPTS='"-Xmx1064m" "-Xms1064m"'
```
Comment (droazen) [inline gradlew:47]: This is not the right place to set this -- probably should use `settings.gradle` instead

File: .gitignore [hunk 2]
```diff
@@ -34,6 +34,12 @@ funcotator_tmp
 #Eclipse files
 /bin/
 .project
+*jar
```
Comment (droazen) [inline .gitignore:37]: `*.jar`

File: src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionIterator.java [hunk 3]
```diff
@@ -119,6 +119,8 @@ private AssemblyRegion loadNextAssemblyRegion() {
         while ( locusIterator.hasNext() && nextRegion == null ) {
             final AlignmentContext pileup = locusIterator.next();
 
+            logger.debug(()->"pileup: " + pileup);
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionIterator.java:122]: Confirm that `toString()` is lazily evaluated here by doing an empirical test.

Add more context info to the logger messages (eg., Class/method name)

File: src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java [hunk 4]
```diff
@@ -1106,4 +1106,15 @@ protected final Object doWork() {
...
     public void closeTool(){
     }
+
+    /**
+     * expose a read-only version of the user-supplied intervals. This can be used by tools that need to explicitly
+     * traverse the intervals themselves (rather than, for example, walking the reads based on the intervals)
+     *
+     * @return - intervals, as an unmodifiable list
+     */
+    protected List<SimpleInterval> getUserIntervals() {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java:1116]: Delete this method, and use `getTraversalIntervals()` instead

**Review by droazen — CHANGES_REQUESTED** [review]
@jamesemery Back to you for some changes

File: src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionIterator.java (hunk 3 above)
Comment (jamesemery) [reply src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionIterator.java:122]: Just made the pileup.toString() throw and exception and the integration tests pass... 

Sure though the pileup toString() method has a lot of details in it anyway.

**Review by droazen — APPROVED** [review]
:+1: merge when ready

**Pull request merged**